modules/oe_migration_workflow/tests/src/Unit/Plugin/migrate/process/SetWorkflowStateTest.php

```diff
@@ -0,0 +1,308 @@
+      ->method('getConfigPrefix')
+      ->willReturn('workflows.workflow');
+
+    $storage = $this->createMock(EntityStorageInterface::class);
+    $storage->expects($this->any())
+      ->method('load')
+      ->willReturnMap(
+        [
+          ['oe_corporate_workflow', $workflow_base],
+          ['test', $workflow_alternative],
+        ]
+      );
+
+    $this->entityTypeManager = $this->createMock(EntityTypeManagerInterface::class);
+    $this->entityTypeManager->expects($this->any())
+      ->method('getDefinition')
+      ->with('workflow')
+      ->willReturn($configEntityTypeInterface);
+    $this->entityTypeManager->expects($this->any())
+      ->method('getStorage')
+      ->with('workflow')
+      ->willReturn($storage);
+
+    $this->configManager = $this->createMock(ConfigManagerInterface::class);
+    $this->configManager->expects($this->any())
+      ->method('getEntityTypeIdByName')
+      ->with($this->logicalOr('workflows.workflow.oe_corporate_workflow', 'workflows.workflow.test'))
+      ->willReturn('workflow');
+  }
+
+  /**
+   * Test de base case without any special configuration.
+   *
+   * @throws \Drupal\migrate\MigrateException
+   * @throws \Exception
+   */
+  public function testTransform() {
+    $this->initializePlugin();
+
+    $value = 'published';
+    $statusDestination = $this->plugin->transform($value, $this->migrateExecutable, $this->row, $this->destinationProperty);
+    $this->assertEquals($statusDestination, $value);
+
+    $value = 'draft';
+    $this->row->setSourceProperty('status', 0);
+    $this->row->setSourceProperty('name', $value);
+    $statusDestination = $this->plugin->transform($value, $this->migrateExecutable, $this->row, $this->destinationProperty);
+    $this->assertEquals($statusDestination, $value);
+  }
+
+  /**
+   * Test that unrecognized workflow status are changed with a new valid status.
+   *
+   * @throws \Drupal\migrate\MigrateException
+   * @throws \Exception
+   */
+  public function testTransformNewWorkflowStatus() {
+
+    $this->initializePlugin();
+
+    $value = 'invalid_and_published';
+    $this->row->setSourceProperty('name', $value);
+    $statusDestination = $this->plugin->transform($value, $this->migrateExecutable, $this->row, $this->destinationProperty);
+    $this->assertEquals('published', $statusDestination);
+
+    $value = 'invalid_and_draft';
+    $this->row->setSourceProperty('name', $value);
+    $this->row->setSourceProperty('status', 0);
+    $statusDestination = $this->plugin->transform($value, $this->migrateExecutable, $this->row, $this->destinationProperty);
+    $this->assertEquals('draft', $statusDestination);
+  }
+
+  /**
+   * Test with a change in the default configuration.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   * @throws \Drupal\migrate\MigrateException
+   */
+  public function testWorkflowConfigAlter() {
```

Comment:
You are not really testing an alter here. You are testing a different workflow.  Not sure it warrants a new test method either. All these tests you can run one after the other in the same test...

Reply:
Yes, maybe I can put it in the same method. Although I don't like to put several tests in the same test method because it's more difficult to debug and find the broken test when it fails.

BTW you're right that the name is not good. It isn't an alter and it's a new/different workflow. I merged it!